py/_io/terminalwriter.py

```diff
@@ -144,12 +144,30 @@ def __init__(self, file=None, stringio=False, encoding=None):
     def fullwidth(self):
         if hasattr(self, '_terminal_width'):
             return self._terminal_width
-        return get_terminal_width()
+        try:
+            width = self._cached_terminal_width
+        except AttributeError:
+            try:
+                signal.signal(signal.SIGWINCH, self._handle_sigwinch)
+            except AttributeError:  # SIGWINCH not supported.
+                width = self._cached_terminal_width = False
```

Comment:
> the cached value is a "global" to begin with, so lets just move the cache out of individual objects up to the module level

Oh OK now I see what you mean. We would have a global cache for the terminal width, which you are right, is global anyway.

@blueyed if you don't mind then I will take this over in a separate PR and test it on Windows anyway. 😉